src/Builder.php

```diff
@@ -64,35 +64,48 @@ public function setOutputDir(?string $outputDir): void
             : static::findOutputDir($this->getBaseDir());
     }
 
-    public static function rebuild(): void
+    public static function rebuild(string $baseDir = null): void
     {
-        $builder = new self(new ConfigFactory(), static::findBaseDir());
+        if ($baseDir === null) {
+            $baseDir = static::findDirContainsComposerJsonRecursively(getcwd());
```

Comment:
The plugin is installed in `$baseDir/vendor/yiisoft/composer-config-plugin`
So the path to the current file is: `$baseDir/vendor/yiisoft/composer-config-plugin/src/Builder.php
So if I'm not mistaken:
```suggestion
            $baseDir = dirname(__DIR__, 4);
```

Reply:
Not necessary. First, [it can be customized](https://getcomposer.org/doc/06-config.md#vendor-dir). Second, it can be symlinked in case yii-dev-tool is used for composer config plugin.

Reply:
IMHO in those cases `$baseDir` must be passed explicitly cause you should not (in some cases even cannot) guess it with random directories scan.
It could be recommended to call rebuild with something like `Builder::rebuild(dirname(__DIR__));`
Maybe change `$baseDir` argument to be required?

Reply:
I've suggested this for a long time ago ;)

Reply:
Alright, let's make it required then.

Reply:
Ok